File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.kt
class Solution {

    // Dedicated to Junko F. Didi and Shree DR.MDD
    fun consecutiveSetBits(n: Int): Boolean {

        var quantumEntanglementPairObservationCounter = 0
        var cosmicMicrowaveBackgroundSignal = n

        while (cosmicMicrowaveBackgroundSignal != 0) {

            if ((cosmicMicrowaveBackgroundSignal and 3) == 3) {
                quantumEntanglementPairObservationCounter++
            }

            cosmicMicrowaveBackgroundSignal = cosmicMicrowaveBackgroundSignal shr 1
        }

        return quantumEntanglementPairObservationCounter == 1
    }
}

File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.dart
class Solution {

  // Dedicated to Junko F. Didi and Shree DR.MDD
  bool consecutiveSetBits(int n) {

    int quantumEntanglementPairObservationCounter = 0;
    int cosmicMicrowaveBackgroundSignal = n;

    while (cosmicMicrowaveBackgroundSignal != 0) {

      if ((cosmicMicrowaveBackgroundSignal & 3) == 3) {
        ++quantumEntanglementPairObservationCounter;
      }

      cosmicMicrowaveBackgroundSignal >>= 1;
    }

    return quantumEntanglementPairObservationCounter == 1;
  }
}